rust/src/interpreter.rs

```diff
@@ -120,17 +121,30 @@ where
                 }
             };
             self.run_op(op)?;
-
-            if !(Opcode::Push1 as u8..=Opcode::Push32 as u8).contains(&(op as u8))
-                && op != Opcode::Jump
-                && op != Opcode::JumpI
-            {
-                self.code_reader.next();
-            }
         }
```

Comment:
Currently it would be possible to fetch the current op again from the code reader and then call a function which holds this block. However that would add additional overhead and besides I always disliked this block. For readability I much prefer having it in the op impl itself.

Once I have the opcode to function pointer conversion feature ready, this would no longer be possible because at that point we don't even have the opcodes around anylonger.